src/Swarm/TUI/View.hs

```diff
@@ -323,11 +327,17 @@ drawWorld g =
                 str [lookupDisplay ((r ^. robotOrientation) >>= toDirection) (r ^. robotDisplay)]
           Nothing -> eWidget
 
--- | Draw a single cell of the world.
-drawCell :: W.Coords -> W.World Int Entity -> (Int, Widget Name)
-drawCell i w = case W.lookupEntity i w of
-  Just e -> (e ^. entityDisplay . displayPriority, displayEntity e)
+-- | Draw a single cell of the world, either hiding those that current robot does not know,
+--   or hiding all/none depending on Left value (True/False).
+drawCell :: W.World Int Entity -> Either Bool Robot -> W.Coords -> (Int, Widget Name)
```

Comment:
How about making a custom data type for this, like `data Hiding = HideAll | HideNone | HideUnknownTo Robot`?  Stuff like `Either Bool` is just confusing, and you always have to read the documentation to remember what `Left True` is supposed to represent and so on.

Reply:
I had exactly those thoughts when clicking "Push" :smile:

Reply:
Thanks for the name, done :heavy_check_mark: